lti_consumer/plugin/views.py

```diff
@@ -704,3 +756,104 @@ def list(self, *args, **kwargs):
                 "error": "above_response_limit",
                 "explanation": "The number of retrieved users is bigger than the maximum allowed in the configuration.",
             }, status=HTTP_403_FORBIDDEN)
+
+
+@csrf_exempt
+@require_http_methods(['POST'])
+def start_proctoring_assessment_endpoint(request):
+    """
+    Receives the Proctoring Tool's message to start the assessment. Emits a signal informing interested parties that
+    the assessment should be started.
+
+    Form Parameters:
+    * JWT (REQUIRED): a signed JWT containing the LTI message
+    """
+    # DECISION: Instead of relying on a config_id URL parameter, let's use the iss claim in signed JWT sent by the
+    #           Proctoring Tool to identify the LtiConfiguration; iss should match the client_id of the
+    #           LtiConfiguration. Although there is a risk that the JWT is not authentic, we will validate the
+    #           authenticity of the JWT and raise an exception if the signature is invalid later in this function.
+    token = request.POST.get('JWT')
+
+    try:
+        jwt = JWT().unpack(token)
+    except BadSyntax:
+        return JsonResponse(
+            {},
+            status=HTTP_400_BAD_REQUEST,
+        )
+
+    jwt_payload = jwt.payload()
+    iss = jwt_payload.get('iss')
+
+    try:
+        lti_config = LtiConfiguration.objects.get(lti_1p3_client_id=iss)
+    except LtiConfiguration.DoesNotExist as exc:
+        log.error("Invalid iss claim '%s' for LTI 1.3 Proctoring Services start_proctoring_assessment_endpoint"
+                  " callback", iss)
+        raise Http404 from exc
+
+    if not lti_config.lti_1p3_proctoring_enabled:
```

Comment:
I _think_ we're intended to interface through the consumer instead of accessing these fields directly (hence the `dl`, `nrps`, and `ags` attributes of the `LtiAdvantageConsumer`). I could use the `proctoring_data` attribute (doesn't roundabout) or check `isinstance(LtiProctoringConsumer, lti_consumer`).

Reply:
no strong opinions here but if we need to build the consumer anyway below I suppose it would be cleaner to just check the type of that consumer object.